knox/views.py

```diff
@@ -16,8 +18,11 @@ class LoginView(APIView):
 
     def post(self, request, format=None):
         if knox_settings.TOKEN_LIMIT is not None:
-            if (request.user.auth_token_set.count() >= knox_settings.TOKEN_LIMIT):
-                return Response({"error": "Maximum amount of tokens allowed per user exceeded."}, status=status.HTTP_401_UNAUTHORIZED)
+            if (request.user.auth_token_set.filter(expires__gt=timezone.now()).count() >= knox_settings.TOKEN_LIMIT):
```

Comment:
unnecessary parenthesis around if statement, also I would write tests for making sure expired token are properly filterered out.